### DISCUSSION TITLE
Extend actor I but I not defined or not a function

### DISCUSSION THREAD

**Discussion by incoming-th** [codeceptjs/CodeceptJS#4241] [Q&A]
Hi,

Here the situation, I try to create custom action for the actor by following the documentation (https://codecept.io/pageobjects/#actor), but I want to be able to call the custom action from another one.

Unfortunately, it errors with "I is not defined", which makes sense:

![image](https://github.com/codeceptjs/CodeceptJS/assets/26375917/ee0422a6-8697-4e14-b98f-77b27cf906ab)

But when trying to call it with "this" it is not working either:

![image](https://github.com/codeceptjs/CodeceptJS/assets/26375917/797f4bb5-aae7-469d-92b5-b4558618e4af)

So I was wondering if I am doing something wrong or if this is just not possible.

Thanks

**Comment by kobenguyent** [comment]
Do you have the same issue with this?

https://github.com/codeceptjs/CodeceptJS/discussions/4091

**Reply by incoming-th** [reply]
Hi,

The issue is not the same, but I did try to look at this and no I do not have any problem with the IDE, both function in the custom actor file are present in autocompletion and no error from the linter/intellisense.

The problem I face is that I cannot call a custom action from another custom action in the "custom_steps.js" file:

![image](https://github.com/codeceptjs/CodeceptJS/assets/26375917/3f9dbd5f-195d-4983-9832-bb9f0537a69c)

**Comment by incoming-th** [comment]
I think I found the cuprite, when refactor as non arrow key function it works:

![image](https://github.com/codeceptjs/CodeceptJS/assets/26375917/ab0a92a3-163d-4e3a-b384-72141cc5d74a)

Closing the thread then.
